### PR TITLE
Add unit test for useFetchAgentFrameworkTraces Hook

### PR DESCRIPTION
**Pull request by Hailong-am**
### Description
Add unit test for useFetchAgentFrameworkTraces

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by gaobinlong — APPROVED** [review]

File: public/hooks/use_fetch_agentframework_traces.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ export const useFetchAgentFrameworkTraces = (traceId: string) => {
       .catch((error) => dispatch({ type: 'failure', error }));
 
     return () => abortController.abort();
```
Comment (wanglam) [inline public/hooks/use_fetch_agentframework_traces.ts:35]: Shall we add one test case for assert this abort method has been called after unmount.

**Review by SuZhou-Joe — APPROVED** [review]

File: public/hooks/use_fetch_agentframework_traces.ts (hunk 1 above)
Comment (Hailong-am) [reply public/hooks/use_fetch_agentframework_traces.ts:37]: sure, case added.

File: public/hooks/use_fetch_agentframework_traces.test.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    });
+
+    expect(services.http.get).toHaveBeenCalledWith(
+      `/api/assistant/trace/${traceId}`,
+      expect.objectContaining({
+        signal: expect.any(Object),
+      })
+    );
+
+    waitFor(() => {
```
Comment (ruanyl) [inline public/hooks/use_fetch_agentframework_traces.test.ts:111]: This seems problematic, if I put `await waitFor`, the test will failed, and also I see an console error on this test

Comment (Hailong-am) [reply public/hooks/use_fetch_agentframework_traces.test.ts]: Add a condition check for unmount to remediate console log error about `[Can't perform a React state update on an unmounted component]`

```typescript
.catch((error) => {
  if (error.name === 'AbortError') return;
    dispatch({ type: 'failure', error });
  });
```

**Review by ruanyl — APPROVED** [review]

**Review by wanglam — APPROVED** [review]

**Pull request merged**